apispec/ext/marshmallow/openapi.py

```diff
@@ -392,6 +393,34 @@ def field2property(self, field, use_refs=True, dump=True, load=True, name=None):
 
         return ret
 
+    def resolve_nested_schema(self, schema, dump, load):
+        """Method to resolve a dictinary from schemas
+
+        Tests to see if the schema is already in the spec and adds it if
+        necessary
+
+        Typically will return a dictionary with the reference to the schema's
+        path in the spec unless the `schema_name_resolver` returns `None` in
+        which case the returned dictoinary will contain the schema's paramaters
+        nested in place
+
+        :param schema: schema to add to the spec
+        :param bool dump: Introspect dump logic.
+        :param bool load: Introspect load logic.
+        """
+
+        schema_cls = resolve_schema_cls(schema)
+        name = self.schema_name_resolver(schema_cls)
+
+        if name and schema_cls not in self.refs:
+            self.spec.components.schema(
+                name,
+                schema=schema,
+                dump=dump,
+                load=load,
+            )
```

Comment:
I don't think we should pass dump/load here. Those are meant to remove some fields when registering a parameter schema or a response schema, but not when registering a definition.

When registering a schema definition, it contains all fields, and dump_only / load_only attributes are documented, which should be enough.

Assuming a schema is used both for parameter and response, it could be auto-registered in dump or load mode depending on who registers it first, which is wrong.

I'd remove `dump` and `load`, and revert the corresponding change in `schema_helper` above.